src/streamlink/plugins/huya.py

```diff
@@ -0,0 +1,43 @@
+import re
+
+from requests.adapters import HTTPAdapter
+
+from streamlink.plugin import Plugin
+from streamlink.plugin.api import http, validate
+from streamlink.stream import HLSStream
+
+USER_AGENT = "Mozilla/5.0 (iPhone; CPU iPhone OS 5_0 like Mac OS X) AppleWebKit/534.46 (KHTML, like Gecko) Version/5.1 Mobile/9A334 Safari/7534.48.3"
+HUYA_URL = "http://m.huya.com/%s"
+
+_url_re = re.compile(r'http(s)?://(www\.)?huya.com/(?P<channel>[^/]+)', re.VERBOSE)
+_hls_re = re.compile(r'^\s*<source\s+src="(?P<url>[^"]+)"/>', re.MULTILINE)
+
+_hls_schema = validate.Schema(
+        validate.all(
+            validate.transform(_hls_re.search),
+            validate.any(
+                None,
+                validate.all(
+                    validate.get('url'),
+                    validate.transform(str)
+                    )
+                )
+            )
+        )
+
+class Huya(Plugin):
+    @classmethod
+    def can_handle_url(self, url):
+        return _url_re.match(url)
+
+    def _get_streams(self):
+        match = _url_re.match(self.url)
+        channel = match.group("channel")
+
+        http.headers.update({"User-Agent": USER_AGENT})
+        http.verify = False
```

Comment:
`m.huya.com uses an invalid security certificate. The certificate is only valid for the following names: webssl.chinanetcenter.com, gcdl0.keke.cn.8686c.com, gcdl1.keke.cn.8686c.com, gcdl2.keke.cn.8686c.com, i.l.inmobicdn.net, storedl0.keke.cn.8686c.com, storedl1.keke.cn.8686c.com, storedl2.keke.cn.8686c.com, *.fn-mart.com, www.1zhe.com, appcdn.liwusj.com, static.liwusj.com, download.liwusj.com, ptlogin.liwusj.com, app.liwusj.com, *.jiedaibao.com, m.jiedaibao.com, *.pinganfang.com, *.anhouse.com, dl.jphbpk.gxpan.cn, dl.givingtales.gxpan.cn, dl.toyblast.gxpan.cn, dl.sds.gxpan.cn, yxhhd.5054399.com, download.ctrip.com, mh.tiancity.com, yxhhd2.5054399.com, app.4399.cn, i.4399.cn, m.4399.cn, a.4399.cn, newsimg.5054399.com, cdn.hxjyios.iwan4399.com, ios.hxjy.iwan4399.com, gjzx.gjzq.com.cn, f.3000test.com, tj.img4399.com, vedio.5054399.com, *.jrjimg.cn, *.zhe800.com, *.tuanimg.com, *.m.zhe800.com, web.im.zhe800.com, message.im.zhe800.com, m.api.zhe800.com, passport.tuan800.com, sso.tuan800.com, *.vxinyou.com, *.gdjh.vxinyou.com, *.3000.com, *.xinstatic.com, www.xin.com, m.xin.com, che.xin.com, pay.game2.cn, *.qiyipic.com, static.iqiyi.com, static.qiyi.com`
The cert was issued by www.chinanetcenter.com, which is a Chinese CDN service provider, but it seems that huya.com doesn't update with the correct cert that has its domain included.